### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by konsumlamm**
Fix some grammar errors, change "modulo" to "modulus" and update links.

### CODE REVIEW

File: Data/Mod.hs [hunk 1]
```diff
@@ -62,21 +62,21 @@ import GHC.TypeNats (Nat, KnownNat, natVal, natVal')
 -- <https://en.wikipedia.org/wiki/Modular_arithmetic#Integers_modulo_n integers modulo m>,
 -- equipped with useful instances.
 --
--- For example, 3 :: 'Mod' 10 stands for the class of integers
--- congruent to \( 3 \bmod 10 \colon \ldots {}−17, −7, 3, 13, 23 \ldots \)
+-- For example, @3 :: 'Mod' 10@ stands for the class of integers
+-- congruent to \( 3 \bmod 10 \colon \ldots, −17, −7, 3, 13, 23, \ldots \)
```
Comment (Bodigrim) [inline Data/Mod.hs]: Please revert both changes.

File: Data/Mod/Word.hs [hunk 2]
```diff
@@ -56,21 +56,21 @@ import GHC.TypeNats (Nat, KnownNat, natVal)
 -- <https://en.wikipedia.org/wiki/Modular_arithmetic#Integers_modulo_n integers modulo m>,
 -- equipped with useful instances.
 --
--- For example, 3 :: 'Mod' 10 stands for the class of integers
--- congruent to \( 3 \bmod 10 \colon \ldots {−17}, −7, 3, 13, 23 \ldots \)
+-- For example, @3 :: 'Mod' 10@ stands for the class of integers
+-- congruent to \( 3 \bmod 10 \colon \ldots, −17, −7, 3, 13, 23, \ldots \)
```
Comment (Bodigrim) [inline Data/Mod/Word.hs]: And this.

File: README.md [hunk 3]
```diff
@@ -37,12 +37,12 @@ in terms of performance. Here is a brief comparison:
 * __Addition.__
   All competing implementations of
   the modular addition involve divisions, while `mod` completely avoids
-  this costly operation. It makes difference even for small numbers;
-  e. g., `sum [1..10^7]` becomes 5x faster. For larger integers the speed up
+  this costly operation. This makes a difference even for small numbers;
+  e.g., `sum [1..10^7]` becomes 5x faster. For larger integers the speed up
```
Comment (Bodigrim) [inline README.md]: Please retain the space inside of e. g.

File: README.md [hunk 4]
```diff
@@ -62,9 +62,9 @@ in terms of performance. Here is a brief comparison:
 * __Overflows.__
   At first glance `modular-arithmetic` is more flexible than `mod`,
   because it allows to specify the underlying representation of a modular residue,
-  e. g., `Mod Integer 100`, `Mod Int 100`, `Mod Word8 100`. We argue that this is
```
Comment (Bodigrim) [inline README.md:65]: And here.

**Review by Bodigrim — COMMENTED** [review]
Thanks, much appreciated!

File: README.md (hunk 3 above)
Comment (konsumlamm) [reply README.md]: Anything I could find online says that it's written "e.g.", do you still want me to revert this change?

File: Data/Mod.hs (hunk 1 above)
Comment (konsumlamm) [reply Data/Mod.hs]: What's wrong with the code block? It renders the expression in monospace, making it clear that it's a Haskell expression.

Comment (Bodigrim) [reply Data/Mod.hs]: Personal aesthetic preferences, I don't like mixing spaces from proportional and monospaced fonts in the same line.

File: README.md (hunk 3 above)
Comment (Bodigrim) [reply README.md]: These are alternative forms and I have a personal preference. 
https://en.m.wiktionary.org/wiki/e._g.#English

File: Data/Mod.hs [hunk 5]
```diff
@@ -63,20 +63,20 @@ import GHC.TypeNats (Nat, KnownNat, natVal, natVal')
 -- equipped with useful instances.
 --
 -- For example, 3 :: 'Mod' 10 stands for the class of integers
--- congruent to \( 3 \bmod 10 \colon \ldots {}−17, −7, 3, 13, 23 \ldots \)
+-- congruent to \( 3 \bmod 10 \colon \ldots −17, −7, 3, 13, 23 \ldots \)
```
Comment (Bodigrim) [inline Data/Mod.hs]: Please leave {} as is.

**Comment by Bodigrim** [comment]
Thanks!

**Pull request merged**